frontend/src/hooks/useGame.ts

```diff
@@ -23,6 +24,9 @@ type State = {
     myDigi5: CardTypeWithId[],
     myBreedingArea: CardTypeWithId[],
 
+    opponentMemory: number,
```

Comment:
Instead of duplicating your states here, you could do something like this:

```ts
type PlayerState = {
    avatar: string,
    name?: string,
    memory: number,
    reveal: CardTypeWithId[],
    hand: CardTypeWithId[],
    deckField: CardTypeWithId[],
    eggDeck: CardTypeWithId[],
    trash: CardTypeWithId[],
    security: CardTypeWithId[],
    tamer: CardTypeWithId[],
    delay: CardTypeWithId[],
    digi1: CardTypeWithId[],
    digi2: CardTypeWithId[],
    digi3: CardTypeWithId[],
    digi4: CardTypeWithId[],
    digi5: CardTypeWithId[],
    breedingArea: CardTypeWithId[],
}

type State = {
    me: PlayerState,
    opponent: PlayerState,
    
    // rest of store, methods and stuff
};
```